runtime/compiler/rpc/J9Client.cpp

```diff
@@ -270,4 +273,44 @@ J9ClientStream::shutdown()
    // destructor is used instead
    }
 
+void
+J9ClientStream::markServerUnreachable(J9JITConfig *jitConfig)
+   {
+   PORT_ACCESS_FROM_JITCONFIG(jitConfig);
+   if (!_serverUnreachable)
+      {
+      _serverUnreachable = true;
+      _localCompStartTime = j9time_current_time_millis();
+      }
+   }
+
+bool
+J9ClientStream::isServerReachable(J9JITConfig *jitConfig, TR::PersistentInfo *info)
+   {
+   if (!_serverUnreachable)
+      return true;
+
+   // check server availability 1 second after the first StreamFailure
+   // if it fails again, the _pingServerInterval will be set to 10 seconds
+   // meaning the server will be pinged every 10 seconds to see if it's back up again
+   // without the 1 second delay after the first StreamFailure would cause the performance of client
+   // to suffer as client could fire off a lot of remote compilations that fails
+   PORT_ACCESS_FROM_JITCONFIG(jitConfig);
+   uint64_t current_time = j9time_current_time_millis();
```

Comment:
I would prefer to use some time routine that exists in the OMR level just in case OMR team want the communication library moved over to them